### PR TITLE
improve e2e stability

### PR DESCRIPTION
**Pull request by edwardfoyle**
## Problem

We've seen an uptick in e2e flakiness related to testing infra and setup. This PR focuses on the failures where the windows e2e tests appear to hang for no reason. 

**Issue number, if available:**
https://github.com/aws-amplify/amplify-backend/issues/1168 and https://github.com/aws-amplify/amplify-backend/issues/1158
(not marking as fixes because I want to evaluate if they continue to occur)
## Changes

1. Split the e2e tests into two jobs, one that runs the deployment tests and another that runs the create-amplify tests. The windows worker seems to fall over when trying to run verdaccio and install from it at the same time, so splitting this up gives the CPU a little more breathing room
2. stagger the start times of the create-amplify tests. This is an attempt to avoid EBUSY on file hot spots when installing packages

**Corresponding docs PR, if applicable:**

## Validation

3 green e2e runs in a row

## Checklist

- [x] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [x] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [x] If this PR requires a docs update, I have linked to that docs PR above.
- [x] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: .github/workflows/health_checks.yml [hunk 1]
```diff
@@ -298,9 +317,10 @@ jobs:
   update_or_publish_versions:
     if: ${{ github.event_name == 'push' && github.ref_name == 'main' }}
     needs:
-      - run_package_manager_e2e_tests
       - test_with_coverage
-      - run_e2e_tests
+      - e2e_package_manager
+      - e2e_deployment
+      - e2e_create_amplify
```
Comment (sobolk) [inline .github/workflows/health_checks.yml:323]: I think it's a good change to distribute failure risk into multiple jobs. This means a probablity of getting green build in 1-3 retries stays flat while we add tests.

However. It sounds like we should think about `split-e2e-tests` + PR check for that (like we have in gen1). If not in this PR we should log GH issue to follow up.

**Review by sobolk — APPROVED** [review]

File: .github/workflows/health_checks.yml (hunk 1 above)
Comment (edwardfoyle) [reply .github/workflows/health_checks.yml:323]: I'd like to go as far as possible using the matrix feature of GH actions before we start generating the workflow files. Reason being I think it makes it much more cumbersome to figure out how / where to add config to the workflow. I'll cut an issue to think more about this as the need arises.

Comment (sobolk) [reply .github/workflows/health_checks.yml:323]: at this point. I'm mostly interested in something that can assert that all tests are included in the workflow.
Which might be less headache than full blown generation.

Comment (edwardfoyle) [reply .github/workflows/health_checks.yml:323]: https://github.com/aws-amplify/amplify-backend/issues/1190

**Review by rtpascual — APPROVED** [review]
LGTM, hoping this does the trick

**Pull request merged**